nixos/modules/services/networking/doh-server.nix

```
@@ -0,0 +1,156 @@
+{
+  config,
+  lib,
+  pkgs,
+  ...
+}:
+
+let
+  cfg = config.services.doh-server;
+  toml = pkgs.formats.toml { };
+  configFile = toml.generate "config.toml" cfg.settings;
+in
+{
+  options.services.doh-server = {
+    enable = lib.mkEnableOption "DNS-over-HTTPS server";
+
+    package = lib.mkPackageOption pkgs "dns-over-https" { };
+
+    settings = lib.mkOption {
+      type = lib.types.submodule {
+        freeformType = toml.type;
+        options = {
+
+          listen = lib.mkOption {
+            type = lib.types.listOf lib.types.str;
+            default = [
+              "127.0.0.1:8053"
+              "[::1]:8053"
+            ];
+            description = "HTTP listen port";
+          };
+
+          localAddr = lib.mkOption {
```

```suggestion
          local_addr = lib.mkOption {
```
Honestly I'd only leave options which need some sensible default values and drop `local_addr`, `cert` and `key` options from here since aren't needed and it's possible to set them via `settings` option without being declared here.